README.md

```diff
@@ -1 +1,293 @@
+> npm install ipfs-repo-migrations
+```
+
+### Use in Node.js
+
+```js
+const migrations = require('ipfs-repo-migrations')
+```
+
+### Use in a browser with browserify, webpack or any other bundler
+
+```js
+const migrations = require('ipfs-repo-migrations')
+```
+
+## Usage
+
+Example:
+
+```js
+const migrations = require('ipfs-repo-migrations')
+const getVersion = require('ipfs-repo-migrations/repo/version')
+
+const repoPath = 'some/repo/path'
+const repoVersion = await getVersion(repoPath)
+
+if(repoVersion < migrations.getLatestMigrationVersion()){
+  // Old repo! Lets migrate to latest version!
+  await migrations.migrate(repoPath)
+}
+```
+
+To migrate your repository using the CLI, see the [how to run migrations](./run.md) tutorial. 
+
+**For tools that build on top of `js-ipfs` and run mainly in the browser environment, be aware that disabling automatic
+migrations leaves the user with no way to run the migrations because there is no CLI in the browser. In such
+a case, you should provide a way to trigger migrations manually.**
+
+### Writing migration
+
+Migrations are one of those things that can be extremely painful on users. At the end of the day, we want users never to have to think about it. The process should be:
+
+- SAFE. No data lost. Ever.
+- Revertible. Tools must implement forward and backward (if possible) migrations.
+- Tests. Migrations have to be well tested.
+- To Spec. The tools must conform to the spec.
+
+If your migration has several parts, it should be fail-proof enough that if one part of migration fails the previous changes
+are reverted before propagating the error. If possible then the outcome should be consistent repo so it migration could
+be run again. 
+
+#### Architecture of migrations
+
+All migrations are placed in the `/migrations` folder. Each folder there represents one migration that follows the migration
+API.
+
+All migrations are collected in `/migrations/index.js`, which should not be edited manually. It is regenerated on
+every run of `jsipfs-migrations add` (manual changes should follow the same style of modifications). 
+**The order of migrations is important and migrations must be sorted in ascending order**.
+
+Each migration must follow this API. It must export an object in its `index.js` that has following properties:
+
+ * `version` (int) - Number that represents the version which the repo will migrate to (eg. `migration-8` will move the repo to version 8).
+ * `description` (string) - Brief description of what the migrations does.
+ * `migrate` (function) - Function that performs the migration (see signature of this function below)
+ * `revert` (function) - If defined then this function will revert the migration to the previous version. Otherwise it is assumed that it is not possible to revert this migration.
+
+##### `migrate(repoPath, isBrowser)`
+
+_Do not confuse this function with the `require('ipfs-repo-migrations').migrate()` function that drives the whole migration process!_
```

Comment:
After reading this, I am now confused. What is the different then?

Reply:
Damn 😂 

There are two functions with the same name in this tool:

 - root one (eq. `require('ipfs-repo-migrations').migrate()`) which is part of the main API, that upon calling will migrate the repository to new version
 - part of "migration API" that is present on the objects exported in each migration for example in `require('ipfs-repo-migrations/migrations/migration-8').migrate()` that only migrate forward the repo by one version

Do you have some idea how to make it more clear?

Reply:
It makes more sense if you move the API to the top, so that is the first thing the user sees. Otherwise it is information overload and the user didn't even get the time to test the module.

Reply:
Yup, I have done that. It is definitely better flow like this...